### PR TITLE
add filter positions users consumer

### PR DESCRIPTION
**Pull request by borovHokage**

### CODE REVIEW

File: src/PositionService.Broker/PositionService.Broker.csproj [hunk 1]
```diff
@@ -3,6 +3,7 @@
   <Import Project="$(ProjectDir)..\..\Common.Configurations.targets" />
 
   <ItemGroup>
+      <!-->ProjectReference Include="..\..\..\DigitalOffice-Models\src\Models\Models.csproj" /-->
```
Comment (MariyaSin) [inline src/PositionService.Broker/PositionService.Broker.csproj:6]: remove it

File: src/PositionService.Data/PositionRepository.cs [hunk 2]
```diff
@@ -74,6 +74,14 @@ public async Task<List<DbPosition>> GetAsync(IGetPositionsRequest request)
...
     }
 
+    public async Task<List<DbPosition>> GetPositionsAsync(List<Guid> positionsIds)
+    {
+      return await _provider.Positions.Where(
+        p => positionsIds.Contains(p.Id)).Include(p => p.Users.Where(u => u.IsActive))
+        .ToListAsync();
+    }
+
+
```
Comment (MariyaSin) [inline src/PositionService.Data/PositionRepository.cs:84]: remove empty line

File: src/PositionService.Data/PositionRepository.cs [hunk 3]
```diff
@@ -74,6 +74,14 @@ public async Task<List<DbPosition>> GetAsync(IGetPositionsRequest request)
       return await dbPosition.ToListAsync();
     }
 
+    public async Task<List<DbPosition>> GetPositionsAsync(List<Guid> positionsIds)
```
Comment (MariyaSin) [inline src/PositionService.Data/PositionRepository.cs:77]: you in positionRepository, do not use the word Position in Method name

Comment (borovHokage) [reply src/PositionService.Data/PositionRepository.cs]: fixed

File: src/PositionService.Data/PositionRepository.cs (hunk 2 above)
Comment (borovHokage) [reply src/PositionService.Data/PositionRepository.cs]: fixed

File: src/PositionService.Broker/Consumers/FilterPositionsUsersConsumer.cs [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+    public async Task<List<PositionFilteredData>> GetPositionFilteredData(IFilterPositionsRequest request)
+    {
+      List<DbPosition> dbPosition = await _repository.GetAsync(request.PositionsIds);
+
+      return dbPosition.Select(
+        pd => new PositionFilteredData(
+          pd.Id,
+          pd.Name,
+          pd.Users.Where(u => u.IsActive).Select(u => u.UserId).ToList())).ToList();
+    }
```
Comment (MariyaSin) [inline src/PositionService.Broker/Consumers/FilterPositionsUsersConsumer.cs:27]: add empty line

File: src/PositionService.Broker/Consumers/FilterPositionsUsersConsumer.cs [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+
+    public async Task<List<PositionFilteredData>> GetPositionFilteredData(IFilterPositionsRequest request)
+    {
+      List<DbPosition> dbPosition = await _repository.GetAsync(request.PositionsIds);
+
+      return dbPosition.Select(
+        pd => new PositionFilteredData(
+          pd.Id,
+          pd.Name,
+          pd.Users.Where(u => u.IsActive).Select(u => u.UserId).ToList())).ToList();
```
Comment (MariyaSin) [inline src/PositionService.Broker/Consumers/FilterPositionsUsersConsumer.cs:26]: remove Users.Where(u => u.IsActive) to repository

File: src/PositionService.Data/PositionRepository.cs [hunk 6]
```diff
@@ -74,6 +74,13 @@ public async Task<List<DbPosition>> GetAsync(IGetPositionsRequest request)
       return await dbPosition.ToListAsync();
     }
 
+    public async Task<List<DbPosition>> GetAsync(List<Guid> positionsIds)
+    {
+      return await _provider.Positions.Where(
```
Comment (MariyaSin) [inline src/PositionService.Data/PositionRepository.cs:79]: move .Where to next line

**Review by MariyaSin — APPROVED** [review]

File: src/PositionService.Data/PositionRepository.cs (hunk 6 above)
Comment (borovHokage) [reply src/PositionService.Data/PositionRepository.cs]: done

File: src/PositionService.Broker/Consumers/FilterPositionsUsersConsumer.cs (hunk 4 above)
Comment (borovHokage) [reply src/PositionService.Broker/Consumers/FilterPositionsUsersConsumer.cs:28]: ok

File: src/PositionService.Broker/Consumers/FilterPositionsUsersConsumer.cs (hunk 5 above)
Comment (borovHokage) [reply src/PositionService.Broker/Consumers/FilterPositionsUsersConsumer.cs]: done

**Review by ndoriomedov — APPROVED** [review]

**Pull request merged**